sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -433,8 +433,14 @@ object DataSourceStrategy {
    */
   protected[sql] def normalizeFilters(
       filters: Seq[Expression],
-      attributes: Seq[AttributeReference]): Seq[Expression] = {
-    filters.filterNot(SubqueryExpression.hasSubquery).map { e =>
```

Comment:
Filtering out subqueries is out of the purpose of `normalizeFilters`. It was not good to implicitly do this. I think we can either updating method document, or moving of filtering of subqueries as a small helper method `filterSubqueryFilters`.

Then we can do `normalizeFilters(filters, attributes)` and `normalizeFilters(filterSubqueryFilters(filters), attributes)`, individually.

Reply:
Done.